### PR TITLE
Add intersection controller lit-labs/observers example

### PR DESCRIPTION
**Pull request by AndrewJakubowicz**
### Context

View here: https://pr977-a6a87b1---lit-dev-5ftespv5na-uc.a.run.app/playground/#sample=examples/observers-intersection-controller

Add a lit-labs/observers intersection controller example.

This sample focusses on observing multiple child elements, and also showing off the core logic that IntersectionController gives you. Essentially we use the intersection controller just to toggle a class on an element when it is in view. Everything else is CSS.

### Testing

Tested manually. Note that this sample is hidden and will be made public in a future PR once all the observers samples are ready.

Thank you!

### CODE REVIEW

**Review by augustjk — APPROVED** [review]
Beautiful!

File: packages/lit-dev-content/samples/examples/observers-intersection-controller/intersection-demo.ts [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+            ></div>`
+        )}
+      </div>
+      <div id="top-line"></div>
+      <div id="bottom-line"></div>
+    `;
+  }
+
+  firstUpdated() {
+    this.intersectionController = new IntersectionController(this, {
```
Comment (sorvell) [inline packages/lit-dev-content/samples/examples/observers-intersection-controller/intersection-demo.ts:38]: Seems like you can follow the normal convention and make this as a class field.

**Review by sorvell — CHANGES_REQUESTED** [review]

File: packages/lit-dev-content/samples/examples/observers-intersection-controller/index.html [hunk 2]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<script type="module" src="./intersection-demo.js"></script>
+<style>
+  * {
```
[the remark this answers is not in the document]
Comment (AndrewJakubowicz) [reply packages/lit-dev-content/samples/examples/observers-intersection-controller/index.html]: Done thank you!

File: packages/lit-dev-content/samples/examples/observers-intersection-controller/intersection-demo.ts (hunk 1 above)
Comment (AndrewJakubowicz) [reply packages/lit-dev-content/samples/examples/observers-intersection-controller/intersection-demo.ts]: Great point! Done!

**Comment by AndrewJakubowicz** [comment]
Decided that this demo should only be merged once we have declarative directive API for the Intersection controller.
This demo doesn't add anything more that an imperative API could already manage. Revisit this demo once a directive based API exists.

**Pull request closed without merging**